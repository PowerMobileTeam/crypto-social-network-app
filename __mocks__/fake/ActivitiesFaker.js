
function load(count) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284",
        getAvatarSource: () => {
          return {
            source:'http://thisisaurl'
          }
        }
      },
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load} 
}

