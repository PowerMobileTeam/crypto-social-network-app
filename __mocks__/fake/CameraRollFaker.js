
function buildPhoto(count) {

  const photos = [];

  return [...Array(count)].map((_, i) => {
    const code = Math.random().toString(16).substring(2).toUpperCase();
    return {
      'node': {
        'timestamp': Date.now(),
        'group_name': 'Camera Roll',
        'type': 'ALAssetTypePhoto',
        'location': {
          'speed': 2.0533344256922823,
          'latitude': 63.5314,
          'heading': 0,
          'longitude': -19.5112,
          'altitude': 0
        },
        'image': {
          'playableDuration': 0,
          'isStored': true,
          'filename': 'IMG_000'+i+'.JPG',
          'width': 3000,
          'height': 2002,
          'uri': 'assets-library://asset/asset.JPG?id=ED7AC36B-A150-4C38-BB8C-'+code+'&ext=JPG'
        }
      }
    }
  });
}

/**
 * getPhotos data faker
 * @param {integer} count
 */
export function getPhotosFaker(count = 3) {
  const photos = buildPhoto(count);
  return {
    'edges': photos,
    'page_info': {
      'has_next_page': false,
      'start_cursor': photos[0].node.image.uri,
      'end_cursor': photos[count-1].node.image.uri
    }
  };
}

