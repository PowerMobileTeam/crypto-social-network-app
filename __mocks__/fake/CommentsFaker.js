
function load(count) {

  let comments = [...Array(count)].map((_, i) => {
    let code = 'guid' +i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284"
      },
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"comment",
    }
  });

  return {
    'comments': comments,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function commentsServiceFaker() {
  return { load: load} 
}

