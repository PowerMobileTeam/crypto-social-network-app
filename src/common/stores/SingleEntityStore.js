import {
  observable,
  action,
  reaction,
  computed
} from 'mobx';

/**
 * Single Entity Store
 */
class SingleEntityStore {

  @observable entity;

  @action
  setEntity(entity) {
    this.entity = entity;
  }
  
  @action
  clear() {
    this.entity = null;
  }
}

export default SingleEntityStore;

