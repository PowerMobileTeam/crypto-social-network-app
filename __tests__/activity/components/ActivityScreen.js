import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import attachmentService from '../../../src/common/services/attachment.service';
import sessionStorage from '../../../src/common/services/session.service';
import ActivityScreen from '../../../src/newsfeed/ActivityScreen';
import UserStore from '../../../src/auth/UserStore';

import SingleEntityStore from '../../../src/common/stores/SingleEntityStore';
import commentsStoreProvider from '../../../src/comments/CommentsStoreProvider';
import {
  FlatList,
  KeyboardAvoidingView
} from 'react-native';


import { commentsServiceFaker } from '../../../__mocks__/fake/CommentsFaker';
import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import CommentsStore from '../../../src/comments/CommentsStore';
import CommentModel from '../../../src/comments/CommentModel';
import Comment from '../../../src/comments/Comment';

let mockEntityStore = function() {
  entity = new SingleEntityStore();
  let entityResponse = activitiesServiceFaker().load(1);
  entity.setEntity(entityResponse.activities[0]);
  return entity;
}

jest.mock('../../../src/capture/CapturePreview', () => 'CapturePreview');
jest.mock('../../../src/newsfeed/ActivityModel', () => 'ActivityModel');
jest.mock('../../../src/comments/CommentModel', () => 'CommentModel');
jest.mock('../../../src/common/BaseModel', () => 'BaseModel');
jest.mock('../../../src/newsfeed/activity/Activity', () => 'Activity');
jest.mock('../../../src/comments/Comment', () => 'Comment');
jest.mock('../../../src/common/components/CenteredLoading', () => 'CenteredLoading');
jest.mock('../../../src/auth/UserStore');
jest.mock('../../../src/common/stores/SingleEntityStore', () => {
  return jest.fn().mockImplementation(() => {
    return mockEntityStore;
  });
});

jest.mock('../../../src/comments/CommentsStore');
jest.mock('../../../src/comments/CommentsStoreProvider');


describe('Activity screen component', () => {

  let user, comments, entity, screen;
  beforeEach(() => {

    let mockResponse = commentsServiceFaker().load(5);
    commentsStoreProvider.get.mockReturnValue({
      comments: mockResponse.comments,
      loadNext: mockResponse['load-next'],
      loadPrevious: mockResponse['load-previous'],
      setText: () => {},
      post: () => {},
      loadComments: () => {
        return mockResponse.comments;
      },
      attachment:{}
    });
    user = new UserStore();
    screen = shallow(
      <ActivityScreen.wrappedComponent user={user}/>
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have a flatlist', async () => {
    screen.update();

    expect(screen.find(FlatList)).toHaveLength(1);
  });

  it('should be wrapped in a keyboard avoiding view', async () => {
    screen.update();

    expect(screen.find(KeyboardAvoidingView)).toHaveLength(1);
  });


  it('should have n comments', async () => {

    let instance = screen.instance();
    const spy = jest.spyOn(instance, 'renderComment');
    screen.update();

    expect(instance.comments.comments.length).toBe(5);
  });


  it('calls post when user comment', async () => {
    screen.update();
    let instance = screen.instance();
    const spy = jest.spyOn(instance, 'postComment');

    const render = screen.dive();
    render.find('TextInput').forEach(child => {
      child.simulate('changeText', 'comment');
    });
    expect(instance.comments.comments.length).toBe(5);
    await render.find('TouchableOpacity').at(1).simulate('press');

    expect(spy).toHaveBeenCalled();
  });
});
