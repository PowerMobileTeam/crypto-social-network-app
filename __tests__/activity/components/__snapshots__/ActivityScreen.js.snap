// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity screen component renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <KeyboardAvoidingView
    behavior="padding"
    enabled={true}
    keyboardVerticalOffset={66}
    style={
      Object {
        "flex": 1,
        "paddingBottom": 0,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <FlatList
        ListEmptyComponent={<CenteredLoading />}
        ListHeaderComponent={
          <Component>
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPress={[Function]}
              style={
                Object {
                  "backgroundColor": "#EEE",
                  "borderRadius": 3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 8,
                  "padding": 8,
                }
              }
              underlayColor="transparent"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#888",
                    "fontSize": 10,
                  }
                }
              >
                 LOAD EARLIER 
              </Text>
            </TouchableHighlight>
          </Component>
        }
        data={
          Array [
            Object {
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "guid0",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "guid": "guid0",
              "mature": false,
              "ownerObj": Object {
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "thumbnail_src": false,
              "type": "comment",
            },
            Object {
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "guid1",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "guid": "guid1",
              "mature": false,
              "ownerObj": Object {
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "thumbnail_src": false,
              "type": "comment",
            },
            Object {
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "guid2",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "guid": "guid2",
              "mature": false,
              "ownerObj": Object {
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "thumbnail_src": false,
              "type": "comment",
            },
            Object {
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "guid3",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "guid": "guid3",
              "mature": false,
              "ownerObj": Object {
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "thumbnail_src": false,
              "type": "comment",
            },
            Object {
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "guid4",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "guid": "guid4",
              "mature": false,
              "ownerObj": Object {
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "thumbnail_src": false,
              "type": "comment",
            },
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={25}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        style={
          Object {
            "backgroundColor": "#FFF",
            "flex": 1,
          }
        }
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
      <Component>
        <Component
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "baseline",
              "backgroundColor": "#FFF",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 5,
            }
          }
        >
          <Image
            source={Object {}}
            style={
              Object {
                "borderColor": "#EEE",
                "borderRadius": 18,
                "borderWidth": 0.5,
                "height": 36,
                "width": 36,
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            autogrow={true}
            editable={true}
            maxHeight={110}
            multiline={true}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSelectionChange={[Function]}
            placeholder="Type your comment..."
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "marginLeft": 8,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
          <Component
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "paddingRight": 8,
                }
              }
            >
              <Icon
                allowFontScaling={false}
                name="md-attach"
                size={24}
                style={
                  Object {
                    "paddingRight": 10,
                  }
                }
              />
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "paddingRight": 8,
                }
              }
            >
              <Icon
                allowFontScaling={false}
                name="md-send"
                size={24}
              />
            </TouchableOpacity>
          </Component>
        </Component>
      </Component>
      <UserAutocomplete
        noFloat={true}
        onSelect={[Function]}
        selection={
          Object {
            "end": 0,
            "start": 0,
          }
        }
      />
    </Component>
    <ActionSheet
      cancelButtonIndex={0}
      onPress={[Function]}
      options={
        Array [
          "Cancel",
          "Gallery",
          "Photo",
          "Video",
        ]
      }
    />
  </KeyboardAvoidingView>
</Component>
`;
